tests/storage/hsm_connect_test.py

```diff
@@ -184,3 +185,16 @@ def test_cache_update(fake_hsm):
     sc = storageServer.ConnectionFactory.connections
     assert sc['1'].connected
     assert hsm.sdCache.knownSDs['sd-uuid-1'] == nfs_find_method
+
+
+def test_empty_conn_list_on_connect(fake_hsm):
+    with pytest.raises(se.InvalidParameterException):
+        fake_hsm.connectStorageServer(sd.ISCSI_DOMAIN, "fake_sp_uuid", [])
+
+
+@pytest.mark.xfail(reason="BZ #2054745, not ready to accept empty connection")
+def test_empty_conn_list_on_disconnect(fake_hsm):
+    # If iSCSI target is used by multiple SDs, engine sends empty connection
+    # list and vdsm has to handle empty list on disconnect.
+    # See https://bugzilla.redhat.com/2054745
+    fake_hsm.disconnectStorageServer(sd.ISCSI_DOMAIN, "fake_sp_uuid", [])
```

Comment:
When faking a uuid, it is little bit nicer to use "fake-sp-uuid". We use this style a lot.